scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java

```diff
@@ -416,30 +428,31 @@ public void copyInputDataToScratch(List<InputSelector> inputSelectors) throws Fi
      * Create the folder hierarchy and select the files to copy
      * from the specified list of FileObjects.
      */
-    private void createFolderHierarchySequentially(
+    protected void createFolderHierarchySequentially(
             DataSpacesFileObject destination, String spaceUri, List<DataSpacesFileObject> spaceFiles,
             Map<String, DataSpacesFileObject> filesToCopy) throws FileSystemException {
 
-        for (DataSpacesFileObject fileObject : spaceFiles) {
-            String relativePath = relativize(spaceUri, fileObject);
+        boolean isDebugEnabled = logger.isDebugEnabled();
+        boolean isFolderHierarchyCreationEnabled = isCreateFolderHierarchySequentiallyEnabled();
+        long startTime = 0;
 
-            DataSpacesFileObject target = destination.resolveFile(relativePath);
+        if (isDebugEnabled) {
+            startTime = System.currentTimeMillis();
+        }
 
-            boolean isDebugEnabled = logger.isDebugEnabled();
+        for (DataSpacesFileObject fileObject : spaceFiles) {
+            String relativePath = relativize(spaceUri, fileObject);
 
-            if (fileObject.isFolder()) {
-                if (isDebugEnabled) {
-                    logger.debug("Creating folder " + target.getRealURI());
-                }
-                target.createFolder();
-                setFolderRightsForRunAsUserMode(target);
-            } else if (fileObject.isFile()) {
-                DataSpacesFileObject parent = target.getParent();
-                if (isDebugEnabled) {
-                    logger.debug("Creating folder " + parent.getRealURI());
+            if (isFolderHierarchyCreationEnabled) {
+                try {
+                    DataSpacesFileObject target = destination.resolveFile(relativePath);
+                    createFolderHierarchy(isDebugEnabled, fileObject, target);
+                } catch (FileSystemException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)